compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -1457,21 +1457,38 @@ trait BCodeBodyBuilder extends BCodeSkelBuilder {
         }
       }
 
-      val methodName = abstractMethod.javaSimpleName
-      val applyN = {
-        val mt = asmMethodType(abstractMethod)
-        mt.toASMType
+      val methodName = samMethod.javaSimpleName
+      val samMethodType = asmMethodType(samMethod).toASMType
+      val needsGenericBridge = samMethodType != instantiatedMethodType
+      val bridgeMethods = atPhase(erasurePhase.prev){
```

Comment:
This can actually be run at `erasurePhase` because when running at a certain phase we see the types as they are before the phase denotation transformer is run.
```suggestion
      val bridgeMethods = atPhase(erasurePhase) {
```